ext/oj/compat.c

```diff
@@ -25,20 +25,8 @@ static void hash_set_cstr(ParseInfo pi, Val kval, const char *str, size_t len, c
         parent->clen      = len;
     } else {
         volatile VALUE rstr = oj_cstr_to_value(str, len, (size_t)pi->options.cache_str);
+        rkey                = oj_calc_hash_key(pi, kval);
```

Comment:
A further optimization would be to remove line 19 then line 21 becomes:
`if (Qundef == val->key_val && Yes == pi->options.create_ok && NULL != pi->options.create_id &&`
and the new line 28 becomes `volatile VALUE rkey                = oj_calc_hash_key(pi, kval);`.

Since oj_calc_hash_key already gets rkey from the parent the suggested change avoids the assignment and stack variable us in the first block.

Reply:
Thanks for your review. I have fixed the codes to follow the suggestion.